include/boost/config/select_compiler_config.hpp

```diff
@@ -92,8 +92,12 @@
 //  MPW MrCpp or SCpp
 #   define BOOST_COMPILER_CONFIG "boost/config/compiler/mpw.hpp"
 
+#elif defined(__IBMCPP__) && defined(__COMPILER_VER__)
```

Comment:
According to https://sourceforge.net/p/predef/wiki/Compilers/ the macro `__COMPILER_VER__` is only defined by the z/OS variant of XL C/C++. I could add `&& !defined(__xlc__) && !defined(__xlC__)` - those macros should only be defined for other XL C/C++ compilers.